Update lldbWebKitTester's CLANG_DEBUG_INFORMATION_LEVEL workaround for workspace builds

emw-apple opened the pull request:
#### 1b9d28989feaef9d3488cdfe4c40795944ac94b8
<pre>
Update lldbWebKitTester&apos;s CLANG_DEBUG_INFORMATION_LEVEL workaround for workspace builds
<a href="https://bugs.webkit.org/show_bug.cgi?id=241863">https://bugs.webkit.org/show_bug.cgi?id=241863</a>

Reviewed by Alexey Proskuryakov.

When building lldbWebKitTester, we used to transparently filter out the
CLANG_DEBUG_INFORMATION_LEVEL setting that buildbot passes. Now that
builds are a single xcodebuild invocation, we cannot do that.

Introduce a WK_OVERRIDE_DEBUG_INFORMATION_LEVEL setting, which
lldbWebKitTester sets to its required debug info level. Change
buildbot&apos;s overrides to prefer this setting over `line-tables-only`.

* Tools/CISupport/build-webkit-org/steps.py:
(CompileWebKit.start):
* Tools/CISupport/ews-build/steps.py:
(CompileWebKit.start):
* Tools/Scripts/webkitdirs.pm:
(buildXCodeProject):
* Tools/lldb/lldbWebKitTester/Configurations/lldbWebKitTester.xcconfig:

Canonical link: <a href="https://commits.webkit.org/251808@main">https://commits.webkit.org/251808@main</a>
</pre>

aproskuryakov commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -2430,8 +2430,9 @@ def start(self):
+            # Some projects (namely lldbWebKitTester) require full debug info, and may override this.
             self.setCommand(self.command + ['DEBUG_INFORMATION_FORMAT=dwarf-with-dsym'])
-            self.setCommand(self.command + ['CLANG_DEBUG_INFORMATION_LEVEL=line-tables-only'])
+            self.setCommand(self.command + ['CLANG_DEBUG_INFORMATION_LEVEL=$(WK_OVERRIDE_DEBUG_INFORMATION_LEVEL:default=line-tables-only)'])
```
How does this work with CMake?

emw-apple replied:
This code is Xcode-only. Just outside the diff context is an `if platform in ('mac', 'ios', 'tvos', 'watchos')` statement.

emw-apple committed:
Update lldbWebKitTester's CLANG_DEBUG_INFORMATION_LEVEL workaround for workspace builds
https://bugs.webkit.org/show_bug.cgi?id=241863

Reviewed by Alexey Proskuryakov.

When building lldbWebKitTester, we used to transparently filter out the
CLANG_DEBUG_INFORMATION_LEVEL setting that buildbot passes. Now that
builds are a single xcodebuild invocation, we cannot do that.

Introduce a WK_OVERRIDE_DEBUG_INFORMATION_LEVEL setting, which
lldbWebKitTester sets to its required debug info level. Change
buildbot's overrides to prefer this setting over `line-tables-only`.

* Tools/CISupport/build-webkit-org/steps.py:
(CompileWebKit.start):
* Tools/CISupport/ews-build/steps.py:
(CompileWebKit.start):
* Tools/Scripts/webkitdirs.pm:
(buildXCodeProject):
* Tools/lldb/lldbWebKitTester/Configurations/lldbWebKitTester.xcconfig:

Canonical link: https://commits.webkit.org/251808@main
